src/com/spikes2212/genericsubsystems/BasicSubsystem.java

```diff
@@ -12,9 +12,11 @@
  *
  * @author Omri "Riki" Cohen
  */
-
 public class BasicSubsystem extends Subsystem {
 
+	/**
+	 * This function should check if the subsystem can move given the speed
+	 */

```

Comment:
change this to something clearer, like "this function when applied to a certain double speed returns true if this subsystem can move at that speed"